From a476e1b6bf6f683bd74000bb30076868c9f92371 Mon Sep 17 00:00:00 2001
From: Bhushan Shah <bhush94@gmail.com>
Date: Wed, 25 Mar 2015 21:10:01 +0530
Subject: [PATCH] Workaround the lockscreen password field focus issue

SDDM theme is also doing this.
Approved by notmart and d__ed

BUG: 344823
FIXED-IN: 5.3
---
 lookandfeel/contents/lockscreen/LockScreen.qml | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/lookandfeel/contents/lockscreen/LockScreen.qml b/lookandfeel/contents/lockscreen/LockScreen.qml
index e48db47..d5cd081 100644
--- a/lookandfeel/contents/lockscreen/LockScreen.qml
+++ b/lookandfeel/contents/lockscreen/LockScreen.qml
@@ -156,6 +156,14 @@ Image {
                             enabled: !authenticator.graceLocked
                             onAccepted: unlockFunction()
                             focus: true
+                            //HACK: Similar hack is needed in sddm loginscreen
+                            //TODO: investigate
+                            Timer {
+                                interval: 200
+                                running: true
+                                repeat: false
+                                onTriggered: passwordInput.forceActiveFocus()
+                            }
                             visible: block.mainItem.model.get(block.mainItem.selectedIndex)["showPassword"]
                             onVisibleChanged: {
                                 if (visible) {

-- 
2.1.4

